### PR TITLE
Document more use cases for Iterator scan()

### PR DESCRIPTION
**Pull request by ve-nt**
This includes explicitly stating that returning `None` from the
closure passed to `scan()` will short the iteration, and that `scan()`
is able to take a mutable reference as it's initial value to mutate
data outside of the iteration.

### CODE REVIEW

File: library/core/src/iter/traits/iterator.rs [hunk 1]
```diff
@@ -1334,6 +1340,67 @@ pub trait Iterator {
     /// assert_eq!(iter.next(), Some(-6));
     /// assert_eq!(iter.next(), None);
     /// ```
+    ///
```
Comment (m-ou-se) [inline library/core/src/iter/traits/iterator.rs:1343]: These examples are quite long, and I'm not sure they really improve understanding in a way that it's worth adding these. Maybe we can just modify the existing example to show stopping early by returning `None` from the closure.

**Review by m-ou-se — COMMENTED** [review]
Thanks for your PR.

I have a few comments:

**Comment by crlf0710** [comment]
@ve-nt Ping from triage, would you mind addressing the comments above? Thanks!

**Comment by ve-nt** [comment]
@m-ou-se Hi, sorry for the delay, I forgot about this PR. 

After reading your perspective, and realizing I had forgotten that a closure can just capture it's environment anyway (so no need to pass in a mutable reference via the initializer argument), I have come to the conclusion that my examples aren't sound. They are more of an abuse of the scan function, and so I don't think they should be documented in the std.

I'm going to close this PR. Again, sorry for taking so long to get back to you on this. Thanks!

**Pull request closed without merging**